src/main/java/de/tum/in/www1/artemis/repository/iris/IrisCourseChatSessionRepository.java

```diff
@@ -0,0 +1,76 @@
+package de.tum.in.www1.artemis.repository.iris;
+
+import java.util.List;
+import java.util.Optional;
+
+import jakarta.validation.constraints.NotNull;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import de.tum.in.www1.artemis.domain.iris.session.IrisCourseChatSession;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * Repository interface for managing {@link IrisCourseChatSession} entities.
+ * Provides custom queries for finding chat sessions based on different criteria.
+ */
+public interface IrisCourseChatSessionRepository extends JpaRepository<IrisCourseChatSession, Long> {
+
+    /**
+     * Finds a list of {@link IrisCourseChatSession} based on the exercise and user IDs.
+     *
+     * @param courseId The ID of the course.
+     * @param userId   The ID of the user.
+     * @return A list of chat sessions sorted by creation date in descending order.
+     */
+    @Query("""
+            SELECT s
+            FROM IrisCourseChatSession s
+            WHERE s.course.id = :courseId
+                AND s.user.id = :userId
+            ORDER BY s.creationDate DESC
+            """)
+    List<IrisCourseChatSession> findByCourseIdAndUserId(@Param("courseId") long courseId, @Param("userId") long userId);
+
+    @Query("""
+                SELECT s
+                FROM IrisCourseChatSession s
+                LEFT JOIN FETCH s.messages
+                WHERE s.course.id = :courseId
+                    AND s.user.id = :userId
+                ORDER BY s.creationDate DESC
+                LIMIT 1
```

Comment:
Please do not use `LIMIT 1` since it does not work with all supported databases. Instead use pageables with size 1 as for example in `getCoverageReportForLatestSolutionSubmissionFromProgrammingExercise()`.

Reply:
@bassner